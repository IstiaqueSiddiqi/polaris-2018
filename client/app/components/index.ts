export * from "./home";
export * from "./about";
export * from "./header";
export * from "./error-page";
