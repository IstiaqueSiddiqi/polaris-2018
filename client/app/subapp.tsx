import * as React from 'react';

interface SubAppProps {
    message: string
}

export class SubApp extends React.Component<SubAppProps, any>{
    render(){
        return(
            <h1>Hello {this.props.message}</h1>
        )
    }
}