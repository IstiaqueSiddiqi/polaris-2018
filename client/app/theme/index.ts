import * as styledComponents from "styled-components";
import { ThemedStyledComponentsModule } from "styled-components";

import ThemeInterface from "./theme";

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider
} = styledComponents as ThemedStyledComponentsModule<ThemeInterface>;

export const theme: ThemeInterface = {
  primaryColor: "#24292e"
};

export { css, injectGlobal, keyframes, ThemeProvider };
export default styled;
