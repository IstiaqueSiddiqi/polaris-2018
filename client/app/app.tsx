import React, { SFC } from "react";
import { Header } from "./components";

export const App: SFC<{}> = props => (
  <div>
    <Header />
    {props.children}
  </div>
);
