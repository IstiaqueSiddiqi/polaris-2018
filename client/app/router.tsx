import React, { SFC } from "react";
import { Switch, BrowserRouter as Router, Route } from "react-router-dom";
import { Home, About } from "./components";
import { StyledApp } from "./app";

export const AppRouter: SFC<{}> = () => (
  <Router>
    <div>
      <Route component={StyledApp} />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route path="/about" component={About} />
      </Switch>
    </div>
  </Router>
);
