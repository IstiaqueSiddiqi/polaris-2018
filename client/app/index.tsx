import * as React from "react";
import * as ReactDOM from "react-dom";
import { AppRouter } from "./router";

/*injectGlobal`
  * { margin: 0; padding: 0; font-family: 'Roboto', sans-serif;}
`;
*/

ReactDOM.render(<AppRouter />, document.getElementById("root"));
