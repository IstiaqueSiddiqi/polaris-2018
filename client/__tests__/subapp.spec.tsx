import React from 'react';
import { SubApp } from '../app/subapp';
import { shallow } from 'enzyme';

test("renders the heading with props supplied", () => {
    const result = shallow(<SubApp message="Polaris"/>).contains(<h1>Hello Polaris</h1>);
    expect(result).toBeTruthy();
});

test("matches snapshot test with no props supplied", () => {
    const snap = shallow(<SubApp message=""/>);
    // Snapshot demo
    expect(snap).toMatchSnapshot();
 });