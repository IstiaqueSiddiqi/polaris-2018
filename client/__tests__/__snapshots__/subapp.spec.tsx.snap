// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot test with no props supplied 1`] = `
<h1>
  Hello 
</h1>
`;

exports[`matching snapshot test 1`] = `
<h1>
  Hello 
</h1>
`;
